product_variant_default_code/models/product.py

```diff
@@ -186,18 +158,43 @@ def _compute_default_code(self):
 class ProductProduct(models.Model):
     _inherit = "product.product"
 
-    manual_code = fields.Boolean(string="Manual Reference")
-
-    @api.model
-    def create(self, vals):
-        product = super(ProductProduct, self).create(vals)
-        if product.reference_mask:
-            render_default_code(product, product.reference_mask)
-        return product
+    manual_code = fields.Boolean(string="Manual Reference", default=False)
+    default_code = fields.Char(
+        compute="_compute_default_code", readonly=False, store=True
+    )
 
-    @api.onchange("default_code")
-    def onchange_default_code(self):
-        self.manual_code = bool(self.default_code)
+    @api.depends(
+        "product_tmpl_id.reference_mask",
+        "product_template_attribute_value_ids.attribute_id.code",
```

Comment:
maybe you do not need to depend on 
product_template_attribute_value_ids.attribute_id.sequence
product_template_attribute_value_ids.attribute_id.name
product_template_attribute_value_ids.product_attribute_value_id.name
as this dependency will change the default mask

as this